WordPress/Classes/ViewRelated/Blog/Blog List/BlogListViewController+SiteCreation.swift

```diff
@@ -2,8 +2,9 @@ extension BlogListViewController {
     @objc
     func launchSiteCreation() {
         let source = "my_sites"
-        JetpackFeaturesRemovalCoordinator.presentSiteCreationOverlayIfNeeded(in: self, source: source, onDidDismiss: {
-            guard JetpackFeaturesRemovalCoordinator.siteCreationPhase() != .two else {
+        let blog = Blog.lastUsedOrFirst(in: ContextManager.sharedInstance().mainContext)
```

Comment:
Fetching the blog seems unnecessary to me, especially since the `lastUsedOrFirst()` function looks computationally heavy. It performs multiple queries and if there are no blogs, we end up doing 3 queries.

Since we only need to know if there are blogs or not, querying for the number of blogs should be enough. Using a query like the one in `PostSignUpInterstitialViewController.numberOfBlogs()` and then passing a boolean `hasBlogs` would simplify things a bit. What do you think?

Reply:
Sounds good.